### DISCUSSION TITLE
x-ref inside x-for does not work?

### DISCUSSION THREAD

**Discussion by KevinArangu** [alpinejs/alpine#1532] [7. Bugs (archived)]
I am trying to put an _x-ref_ inside a  template with x-for  and the console gives me an error. 

(Console Error: cdn.min.js:1 Alpine Expression Error: Cannot read property '_x_refs' of undefined)

here is my example:

![clg](https://user-images.githubusercontent.com/38361709/122294572-300df680-cec6-11eb-80bd-2747cb4afcc8.png)

![img](https://user-images.githubusercontent.com/38361709/122294575-300df680-cec6-11eb-998f-8d18dfb40665.png)

**Comment by KevinBatdorf** [comment]
`x-ref="i"` never actually worked, but in V2 you could do `:x-ref="i"` and bind that as a ref. That works in V2 because the refs were looked up on demand when invoked. V3 will add them to an object. 

I don't know if there's a clean way to do this, but read the responses here for a couple of ideas: https://github.com/alpinejs/alpine/issues/1517

**Reply by KevinArangu** [reply]
in v2 if it worked, I had already done it in my web application. I'm just migrating from alpine v2 to v3. I will look for another way to do the same. I hope they add that feature soon. Thank you so much.

**Reply by KevinBatdorf** [reply]
In V2 that code wouldn't error, but it also didn't work as you might expect. You're assigning the string i as the ref so you would access it $refs.i which is likely not what you want. You can check the devtools to see what's rendered. 

Since scoping has changed in V3 you might be able to just put x-data on that li element and make it work. Depends on the rest of your code though. If you put a demo up somewhere I can help find a solution if you're still stuck refactoring it.
